### PR TITLE
Convert reddit.com session to m.reddit.com session

### PR DESCRIPTION
**Pull request by ajacksified**
:eyeglasses: @kemitche and @JordanMilne and @curioussavage

### CODE REVIEW

File: src/server.jsx [hunk 1]
```diff
@@ -293,6 +303,41 @@ class Server {
...
+      if (skipAuth(app, this.url)) {
+        yield next;
+        return;
+      }
+
+       let session = this.cookies.get('reddit_session');
+
+      if (!this.token &&
+          !this.cookies.get('token') &&
+          this.cookies.get('reddit_sesion')) {
```
Comment (kemitche) [inline src/server.jsx]: sesion?

File: src/routes.jsx [hunk 2]
```diff
@@ -228,6 +228,7 @@ function routes(app) {
     var page;
     var sort = this.query.sort || 'hot';
     var ctx = this;
+    console.log('index', ctx.token);
```
Comment (JordanMilne) [inline src/routes.jsx]: nit: is this still needed?

**Comment by JordanMilne** [comment]
How does this handle logging out? Should we just disable that button? Also, should we detect a logout / account switch on .reddit.com and handle it accordingly?

**Comment by ajacksified** [comment]
Log out will only delete the mweb token - it won't delete the reddit desktop session. Probably should, I'll update.

This is mostly to handle an influx of users coming in from the desktop view; I'm not attempting to handle people going back and forth. I guess I could set a second cookie, like `original_session`, to see if that matches and re-log if not? I'll think about that and possibly handle it in a second pass.

**Comment by JordanMilne** [comment]
IIRC the session cookie has the base10 user id at the start it so you can parse the base36 user id out of the current user's `fullname` and compare the two, but this looks good for now :fish:

**Pull request merged**